src/extension.ts

```diff
@@ -469,9 +473,20 @@ export function activate(context: vscode.ExtensionContext) {
   azdata.dataprotocol.registerObjectExplorerProvider(objectExplorer);
 
   registerHomeDashboardTabs(context, appContext);
+
+  // create telemetry reporter on extension activation
+  appContext.reporter = new TelemetryReporter(
+    Telemetry.extensionId,
+    Telemetry.extensionVersion,
+    Telemetry.instrumentationKey
+  );
+  // ensure it gets property disposed
+  context.subscriptions.push(appContext.reporter);
 }
 
 // export let objectExplorer:azdata.ObjectExplorerProvider | undefined; // TODO should we inject this instead?
 
 // this method is called when your extension is deactivated
-export function deactivate() {}
+export function deactivate() {
+  appContext.reporter?.dispose();
```

Comment:
You can just add the reporter to context.subscriptions to have it automatically disposed when the extension is deactivated.